src/main/resources/category/java/remoting.xml

```diff
@@ -408,535 +408,584 @@ return HttpClientBuilder.create()
+    <priority>2</priority>
+    <properties>
+    <property name="tags" value="jpinpoint-rule,multi-threading" type="String" description="classification"/>
+    <property name="xpath">
     <value><![CDATA[
-(: exclude method annotated with PostConstruct :)
-//ClassOrInterfaceBodyDeclaration[count(Annotation//Name[@Image='PostConstruct']) = 0]
-(: exclude factory where config is in method returning 'this' and one method, typically build(), returning objectMapper :)
-/MethodDeclaration[count(Block//ReturnStatement/Expression/PrimaryExpression/PrimaryPrefix[@ThisModifier=true()])=0
-or ancestor::ClassOrInterfaceBody[count(.//MethodDeclaration/ResultType/Type[pmd-java:typeIs('com.fasterxml.jackson.databind.ObjectMapper')]) != 1]]
-(: modify operations on objectMapper in a method where objectMapper is a formal parameter or field :)
-//Statement//Name[(concat(substring-before(@Image, '.config'), substring-before(@Image, '.set'), substring-before(@Image, '.register')
-, substring-before(@Image, '.find'), substring-before(@Image, '.disable'), substring-before(@Image, '.enable'), substring-before(@Image, '.activate')
-, substring-before(@Image, '.deactivate'), substring-before(@Image, '.add'), substring-before(@Image, '.clear'))
-=
-(ancestor::MethodDeclaration//FormalParameter|ancestor::ClassOrInterfaceBody//FieldDeclaration)[pmd-java:typeIs('com.fasterxml.jackson.databind.ObjectMapper')]//VariableDeclaratorId/@Name)
-]
-,
-(: modify operations on objectMapper in a constructor where objectMapper is a formal parameter or static field :)
-//ConstructorDeclaration//Statement//Name[(concat(substring-before(@Image, '.config'), substring-before(@Image, '.set'), substring-before(@Image, '.register')
-, substring-before(@Image, '.find'), substring-before(@Image, '.disable'), substring-before(@Image, '.enable'), substring-before(@Image, '.activate')
-, substring-before(@Image, '.deactivate'), substring-before(@Image, '.add'), substring-before(@Image, '.clear'))
-=
-(ancestor::ConstructorDeclaration//FormalParameter|ancestor::ClassOrInterfaceBody//FieldDeclaration[@Static=true()])[pmd-java:typeIs('com.fasterxml.jackson.databind.ObjectMapper')]//VariableDeclaratorId/@Name)
-]
-]]>
+    (: exclude method annotated with PostConstruct :)
+    //ClassOrInterfaceBodyDeclaration[count(Annotation//Name[@Image='PostConstruct']) = 0]
+    (: exclude factory where config is in method returning 'this' and one method, typically build(), returning objectMapper :)
+    /MethodDeclaration[count(Block//ReturnStatement/Expression/PrimaryExpression/PrimaryPrefix[@ThisModifier=true()])=0
+    or ancestor::ClassOrInterfaceBody[count(.//MethodDeclaration/ResultType/Type[pmd-java:typeIs('com.fasterxml.jackson.databind.ObjectMapper')]) != 1]]
+    (: modify operations on objectMapper in a method where objectMapper is a formal parameter or field :)
+    //Statement//Name[(concat(substring-before(@Image, '.config'), substring-before(@Image, '.set'), substring-before(@Image, '.register')
```

Comment:
FYI: I created a compact way for similar construction: 
```
and matches(@MethodName,"(set|put|clear|add|remove|replace)")
```

Reply:
ah, die is handig ja. denk ik niet vaak aan regex gebruiken. Deze regel is is nog todo, neem ik dan mee.